src/aiida/manage/configuration/config.py

```diff
@@ -579,9 +579,18 @@ def delete_profile(self, name: str, delete_storage: bool = True) -> None:
 
         if delete_storage:
             storage_cls = StorageFactory(profile.storage_backend)
-            storage = storage_cls(profile)
-            storage.delete()
-            LOGGER.report(f'Data storage deleted, configuration was: {profile.storage_config}')
+            if profile.storage_backend == 'core.sqlite_zip' and not Path(profile.storage_config['filepath']).exists():
```

Comment:
`storage_config` is of type `dict[str, t.Any]` so you can have a key error here and below